lib/chef/knife/bmcs_server_show.rb

```diff
@@ -22,16 +36,61 @@ class BmcsServerShow < Knife
              long: '--instance_id LIMIT',
              description: 'The OCID of the server to display. (required)'
 
+      def lookup_compartment_name(compartment_id)
+        compartments = identity_client.list_compartments(bmcs_config.tenancy, {})
+        compartments.data && compartments.data.each do |compartment|
+          next unless compartment.id == compartment_id
+          return compartment.description || compartment.name
+        end
+      end
+
+      def lookup_image_name(image_id)
+        images = compute_client.list_images(compartment_id, {})
+        images.data && images.data.each do |image|
+          next unless image.id == image_id
+          return image.display_name
+        end
+      end
+
+      def lookup_vcn(compartment_id)
+        vcns = network_client.list_vcns(compartment_id, {})
```

Comment:
OK, I made some changes here, before I saw you updated the comment.  I figured out that the vcn_id could be piggybacked up in the vnic info.  I just use the vcn_id from the primary interface, rather than looking up the vcns from the compartment.

I'm not sure that is optimal or not...